ui/src/flux/components/FluxHeader.tsx

```diff
@@ -1,60 +1,70 @@
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
 
 import FluxOverlay from 'src/flux/components/FluxOverlay'
-import {OverlayContext} from 'src/shared/components/OverlayTechnology'
+import OverlayTechnology from 'src/reusable_ui/components/overlays/OverlayTechnology'
 import PageHeader from 'src/shared/components/PageHeader'
-import {
-  showOverlay,
-  ShowOverlayActionCreator,
-} from 'src/shared/actions/overlayTechnology'
 
 import {Service} from 'src/types'
 
 interface Props {
-  showOverlay: ShowOverlayActionCreator
   service: Service
 }
 
-class FluxHeader extends PureComponent<Props> {
+interface State {
+  showOverlay: boolean
+}
+
+class FluxHeader extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      showOverlay: false,
+    }
+  }
+
   public render() {
     return (
-      <PageHeader
-        titleText="Flux Editor"
-        fullWidth={true}
-        optionsComponents={this.optionsComponents}
-      />
+      <>
+        <PageHeader
+          titleText="Flux Editor"
+          fullWidth={true}
+          optionsComponents={this.optionsComponents}
+        />
+        {this.renderOverlay}
+      </>
     )
   }
 
+  private handleToggleOverlay = (): void => {
+    this.setState({showOverlay: !this.state.showOverlay})
+  }
+
   private get optionsComponents(): JSX.Element {
     return (
-      <button onClick={this.overlay} className="btn btn-sm btn-default">
+      <button
+        onClick={this.handleToggleOverlay}
+        className="btn btn-sm btn-default"
+      >
         Edit Connection
       </button>
     )
   }
 
-  private overlay = () => {
+  private get renderOverlay(): JSX.Element {
     const {service} = this.props
```

Comment:
since this is a getter, it doesn't necessarily need to be named `renderOverlay`; it could just be called `overlay` and make sense. but either way is fine. i think i saw this another place as well